lib/ash_json_api/json_schema/open_api.ex

```diff
@@ -0,0 +1,956 @@
+    {path, {route.method, operation}}
+  end
+
+  @spec operation(Route.t(), resource :: module, path_params :: [String.t()]) ::
+          Operation.t()
+  defp operation(route, resource, path_params) do
+    unless path_params == [] or path_params == ["id"] do
+      raise "Haven't figured out more complex route parameters yet."
+    end
+
+    action = Ash.Resource.Info.action(resource, route.action)
+
+    %Operation{
+      description:
+        action.description ||
+          "#{action.name} operation on #{AshJsonApi.Resource.Info.type(resource)} resource",
+      tags: [to_string(AshJsonApi.Resource.Info.type(resource))],
+      parameters: path_parameters(path_params, action) ++ query_parameters(route, resource),
+      responses: %{
+        :default => %Reference{
+          "$ref": "#/components/responses/errors"
+        },
+        200 => response_body(route, resource)
+      },
+      requestBody: request_body(route, resource)
+    }
+  end
+
+  @spec path_parameters(path_params :: [String.t()], action :: Actions.action()) ::
+          [Parameter.t()]
+  defp path_parameters(path_params, action) do
+    Enum.map(path_params, fn param ->
+      description =
+        action.arguments
+        |> Enum.find(&(to_string(&1.name) == param))
+        |> case do
+          %{description: description} when is_binary(description) -> description
+          _ -> nil
+        end
+
+      %Parameter{
+        name: param,
+        description: description,
+        in: :path,
+        required: true,
+        schema: %Schema{type: :string}
+      }
+    end)
+  end
+
+  @spec query_parameters(
+          Route.t(),
+          resource :: module
+        ) :: [Parameter.t()]
+  defp query_parameters(%{type: :index} = route, resource) do
+    [
+      filter_parameter(resource),
+      sort_parameter(resource),
+      page_parameter(),
+      include_parameter(),
+      fields_parameter(resource)
+    ] ++
+      read_argument_parameters(route, resource)
+  end
+
+  defp query_parameters(%{type: type}, _resource)
+       when type in [:post_to_relationship, :patch_relationship, :delete_from_relationship] do
+    []
+  end
+
+  defp query_parameters(%{type: type} = route, resource) when type in [:get, :related] do
+    [include_parameter(), fields_parameter(resource)] ++ read_argument_parameters(route, resource)
+  end
+
+  defp query_parameters(_route, resource) do
+    [include_parameter(), fields_parameter(resource)]
+  end
+
+  @spec filter_parameter(resource :: module) :: Parameter.t()
+  defp filter_parameter(resource) do
+    %Parameter{
+      name: :filter,
+      in: :query,
+      description:
+        "Filters the query to results with attributes matching the given filter object",
+      required: false,
+      style: :deepObject,
+      schema: filter_schema(resource)
+    }
+  end
+
+  @spec sort_parameter(resource :: module) :: Parameter.t()
+  defp sort_parameter(resource) do
```

Comment:
The sort can also be a string in the form of `foo,-bar,baz` (as per json api spec).